File: Data/Models/TransferNote.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Nebula.Data.Models
{
    public class TransferNote
    {
        public int Id { get; set; }

        /// <summary>
        /// Id Almacén de Origen.
        /// </summary>
        public int? OriginId { get; set; }

        /// <summary>
        /// Almacén Origen.
        /// </summary>
        [ForeignKey("OriginId")]
        public Warehouse Origin { get; set; }

        /// <summary>
        /// Id Almacén de destino.
        /// </summary>
        public int? TargetId { get; set; }

        /// <summary>
        /// Almacén Destino.
        /// </summary>
        [ForeignKey("TargetId")]
        public Warehouse Target { get; set; }

        /// <summary>
        /// Motivo de Inventario.
        /// </summary>
        [MaxLength(250)]
        public string Motivo { get; set; }

        /// <summary>
        /// Fecha de registro.
        /// </summary>
        [DataType(DataType.Date)]
        public DateTime StartDate { get; set; }

        /// <summary>
        /// Observación.
        /// </summary>
        [MaxLength(250)]
        public string Remark { get; set; }

        /// <summary>
        /// Estado de la Transferencia.
        /// </summary>
        [MaxLength(250)]
        public string Status { get; set; }

        /// <summary>
        /// Año de Inventario.
        /// </summary>
        [MaxLength(250)]
        public string Year { get; set; }

        /// <summary>
        /// Mes de Inventario.
        /// </summary>
        [MaxLength(250)]
        public string Month { get; set; }

        /// <summary>
        /// Detalles Nota de Trasferencia.
        /// </summary>
        public List<TransferNoteDetail> TransferNoteDetails { get; set; }
    }
}

File: Angular/src/app/inventory/interfaces/transfer-note.ts
import {TransferNoteDetail} from './transfer-note-detail';
import {Warehouse} from '../../system/interfaces';

export class TransferNote {
  constructor(
    public id: number | any = null, //
    public origin: Warehouse | any = undefined, // Almacén Origen.
    public originId: number = 0, // Id Almacén de Origen.
    public target: Warehouse | any = undefined, // Almacén Destino.
    public targetId: number = 0, // Id Almacén de destino.
    public motivo: string = '', // Motivo de Inventario.
    public startDate: any = null, // Fecha de registro.
    public status: string = '', // Estado de la Transferencia.
    public transferNoteDetails: Array<TransferNoteDetail> = new Array<TransferNoteDetail>()) {
  }
}
